### PR TITLE
Added proxy guide for nextjs

### PR DESCRIPTION
**Pull request by sp3n**

### CODE REVIEW

File: docs/proxy/guides/nextjs.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          }
+      ];
+  },
+}
+```
+
+Choose a generic or irrelevant name for the subdirectory. If you choose something like `analytics` or `plausible`,
+it might get blocked in the future.
+
+Using our extensions such as hash-based routing, page exclusions or outbound link click tracking? Edit your `next.config.js` and change the name from `script.js` to the script you want to use: `script.hash.js`, `script.exclusions.js` or `script.outbound-links.js`. Want to use more than one extension? You can chain them like this: `script.hash.exclusions.outbound-links.js`
```
Comment (ukutaht) [inline docs/proxy/guides/nextjs.md:51]: Changing the name of `script.js` wouldn't have any effect. To use these extensions, one should change from `plausible.js` to `plausible.outbound-links.js` in the `destination: ` field.

**Comment by ukutaht** [comment]
Looks good, I think there's a small issue with the extensions but everything else looks good.

**Comment by ukutaht** [comment]
Amazing, thanks!

**Comment by ukutaht** [comment]
Just to confirm, is this intended to work on Vercel? So if we link to it from the main page, would it be appropriate to name it `Next.JS / Vercel`?

**Comment by sp3n** [comment]
> Just to confirm, is this intended to work on Vercel? So if we link to it from the main page, would it be appropriate to name it `Next.JS / Vercel`?

Indeed, I wrote the guide while doing it for a site built using Next.JS that is hosted on Vercel: [rhythmtowers.com](https://rhythmtowers.com)

**Comment by metmarkosaric** [comment]
thanks @sp3n! i've renamed it to mention Vercel and added a note about your contribution too

**Comment by alvarlagerlof** [comment]
Hi, so this does not seem to work for me. First of all, Next.js 10.2.3 complains about there not being a permanent: bool property in the redirect object, second, it doesn't seem to proxy the file at all for me.

image
image

My code is like this
```js
const withMDX = require("@next/mdx")({
  extension: /\.mdx$/,
});

module.exports = withMDX({
  pageExtensions: ["js", "jsx", "mdx"],
  async redirects() {
    return [
      {
        source: "/js/script.js",
        destination: "https://plausible.io/js/plausible.outbound-links.js",
        permanent: true,
      },
      {
        source: "/api/event",
        destination: "https://plausible.io/api/event",
        permanent: true,
      },
    ];
  },
});
```

```html
 <script
  defer
  data-domain="neurodiversity.wiki"
  src="/js/script.js"
></script>
```

**Comment by alvarlagerlof** [comment]
Nevermind, I used redirects instead of rewrites

**Pull request merged**